erigon-lib/diagnostics/client.go

```diff
@@ -101,8 +105,64 @@ func (d *DiagnosticClient) Setup() {
 	d.setupBodiesDiagnostics(rootCtx)
 	d.setupResourcesUsageDiagnostics(rootCtx)
 	d.setupSpeedtestDiagnostics(rootCtx)
+	d.runSaveProcess(rootCtx)
+	d.runStopNodeListener(rootCtx)
 
 	//d.logDiagMsgs()
+
+}
+
+func (d *DiagnosticClient) runStopNodeListener(rootCtx context.Context) {
+	go func() {
+		ch := make(chan os.Signal, 1)
+		signal.Notify(ch, os.Interrupt, syscall.SIGTERM)
+		select {
+		case <-ch:
+			d.SaveData()
+		case <-rootCtx.Done():
+		}
+	}()
+}
+
+// Save diagnostic data by time interval to reduce save events
+func (d *DiagnosticClient) runSaveProcess(rootCtx context.Context) {
+	ticker := time.NewTicker(5 * time.Minute)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				d.SaveData()
+			case <-rootCtx.Done():
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (d *DiagnosticClient) SaveData() {
+	err := d.db.Update(d.ctx, func(tx kv.RwTx) error {
+		err := SnapshotDownloadUpdater(d.syncStats.SnapshotDownload)(tx)
```

Comment:
Range over list of 'updater's instead may be easier to extend

Reply:
done